src/ServicePulse.Host/vue/src/components/DataView.vue

```diff
@@ -1,33 +1,26 @@
-<script setup>
+<script setup lang="ts" generic="T">
 import { onMounted, onUnmounted, ref, watch } from "vue";
-import { useFetchFromServiceControl } from "../composables/serviceServiceControlUrls";
+// eslint-disable-next-line vue/prefer-import-from-vue
+import type { UnwrapRefSimple } from "@vue/reactivity";
+import { useTypedFetchFromServiceControl } from "../composables/serviceServiceControlUrls";
 import ItemsPerPage from "../components/ItemsPerPage.vue";
 import PaginationStrip from "../components/PaginationStrip.vue";
-const props = defineProps({
-  apiUrl: String,
-  itemsPerPageOptions: {
-    type: Array,
-    default: () => [20, 35, 50, 75],
-  },
-  itemsPerPage: {
-    type: Number,
-    default: 50,
-  },
-  autoRefreshSeconds: {
-    type: Number,
-  },
-  showPagination: {
-    type: Boolean,
-    default: true,
-  },
-  showItemsPerPage: {
-    type: Boolean,
-    default: false,
-  },
-});
 
-const refreshTimer = ref();
-const items = ref([]);
+const props = withDefaults(
+  defineProps<{
+    itemsType: T[];
```

Comment:
Would it make sense to turn this into a `v-model`? That way the caller would at least have access to the items if they needed them for some reason and the prop would not be completely irrelevant? Perhaps that would mean it is no longer necessary to pass the items to the slot. If we include total-count in the v-model then we would not need to pass anything in to the footer either.

Reply:
I am not 100% sure what you mean, @mikeminutillo .
Happy for you to refractor?